## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#e3226f7a-1a26-432b-91f0-f3536ad11bb3) |
| mint-compression-fs.sh | :x: [more...](#4e4bf975-08eb-4e78-a60d-927c2a146d1d) |
<a name="e3226f7a-1a26-432b-91f0-f3536ad11bb3"></a>
### 8282-c7037e1/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:31754
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 9e0b47818224:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 3 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... FAILED in 5 minutes and 5 seconds
{
  "name": "aws-sdk-php",
  "function": "getObject ( array $params = [] )",
  "args": {
    "Bucket": "aws-sdk-php-5d8516f6cd66c",
    "Object": "obj1"
  },
  "duration": "47",
  "status": "FAIL",
  "error": "Error executing \"PutObject\" on \"http://72.28.97.59:31754/aws-sdk-php-5d8516f6cd66c/obj1\"; AWS HTTP error: Client error: `PUT http://72.28.97.59:31754/aws-sdk-php-5d8516f6cd66c/obj1` resulted in a `400 Bad Request` response:\n<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>IncompleteBody</Code><Message>You did not provide the number of byte (truncated...)\n IncompleteBody (client): You did not provide the number of bytes specified by the Content-Length HTTP header. - <?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>IncompleteBody</Code><Message>You did not provide the number of bytes specified by the Content-Length HTTP header.</Message><Key>obj1</Key><BucketName>aws-sdk-php-5d8516f6cd66c</BucketName><Resource>/aws-sdk-php-5d8516f6cd66c/obj1</Resource><RequestId>15C6386C9A16953A</RequestId><HostId>114300f6-5f4c-45a9-9bba-885dcc418c31</HostId></Error>"
}

Executed 2 out of 14 tests successfully.
```
<a name="4e4bf975-08eb-4e78-a60d-927c2a146d1d"></a>
### 8282-c7037e1/mint-compression-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:30967
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 0855fd6e6202:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 2 seconds
(5/14) Running awscli tests ... done in 1 minutes and 55 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 39 seconds
(8/14) Running minio-dotnet tests ... done in 27 seconds
(9/14) Running minio-go tests ... done in 40 seconds
(10/14) Running minio-java tests ... done in 49 seconds
(11/14) Running minio-js tests ... done in 48 seconds
(12/14) Running minio-py tests ... FAILED in 1 minutes and 52 seconds
{
  "message": "AccessDenied: message: Access Denied",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1341, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2036, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1355, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "args": {
    "bucket_name": "minio-py-test-137d7d39-80ff-4af2-8289-6b1b72bf0d40",
    "response.reason": "Forbidden",
    "response.status": 403,
    "response._body": "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>AccessDenied</Code><Message>Request has expired</Message><Key>bcb4f77f-adac-44cc-943a-1e8917d2cd48</Key><BucketName>minio-py-test-137d7d39-80ff-4af2-8289-6b1b72bf0d40</BucketName><Resource>/minio-py-test-137d7d39-80ff-4af2-8289-6b1b72bf0d40/bcb4f77f-adac-44cc-943a-1e8917d2cd48</Resource><RequestId>15C638941B5C67FD</RequestId><HostId>9543ac31-b536-403c-b9cb-1ac823f09b43</HostId></Error>",
    "presigned_get_object_url": "http://72.28.97.59:30967/minio-py-test-137d7d39-80ff-4af2-8289-6b1b72bf0d40/bcb4f77f-adac-44cc-943a-1e8917d2cd48?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20190920%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20190920T182203Z&X-Amz-Expires=5&X-Amz-SignedHeaders=host&X-Amz-Signature=c67e54797a8a368271f1b3517f7ea004249742324bd7e579f8093272da858aec",
    "object_name": "bcb4f77f-adac-44cc-943a-1e8917d2cd48",
    "response.headers": "{\"_container\": {\"accept-ranges\": [\"Accept-Ranges\", \"bytes\"], \"content-length\": [\"Content-Length\", \"440\"], \"content-security-policy\": [\"Content-Security-Policy\", \"block-all-mixed-content\"], \"content-type\": [\"Content-Type\", \"application/xml\"], \"server\": [\"Server\", \"MinIO/DEVELOPMENT.2019-09-20T18-01-17Z\"], \"vary\": [\"Vary\", \"Origin\"], \"x-amz-request-id\": [\"X-Amz-Request-Id\", \"15C638941B5C67FD\"], \"x-xss-protection\": [\"X-Xss-Protection\", \"1; mode=block\"], \"date\": [\"Date\", \"Fri, 20 Sep 2019 18:22:08 GMT\"]}}"
  },
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "duration": 61,
  "status": "FAIL",
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)"
}

Executed 11 out of 14 tests successfully.
```

--- a/.dockerignore
+++ b/.dockerignore
@@ -1,0 +1,2 @@
+.git
+.github

--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,9 @@ LDFLAGS := $(shell go run buildscripts/gen-ldflags.go)
 GOARCH := $(shell go env GOARCH)
 GOOS := $(shell go env GOOS)
 
-TAG ?= $(USER)
+VERSION ?= $(shell git describe --tags)
+TAG ?= "minio/minio:$(VERSION)"
+
 BUILD_LDFLAGS := '$(LDFLAGS)'
 
 all: build

